src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -0,0 +1,222 @@
+package org.corfudb.infrastructure;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.runtime.view.LayoutView;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Instantiates and performs failure detection and handling asynchronously.
+ * <p>
+ * Failure Detector:
+ * Executes detection policy (eg. PeriodicPollingPolicy).
+ * It then checks for status of the nodes. If the result map is not empty,
+ * there are failed nodes to be addressed. This then triggers the failure
+ * handler which then responds to these failures based on a policy.
+ * <p>
+ * Created by zlokhandwala on 9/28/16.
+ */
+@Slf4j
+public class ManagementServer extends AbstractServer {
+
+    @Override
+    public void reset() {
+    }
+
+    @Override
+    public void reboot() {
+    }
```

Comment:
There are other things to be handled as well apart from just the new policy, I have in mind.
So I thought of pushing the reboot and reset functions to the next PR.